libs/map/depth_point_map.cpp

```
@@ -0,0 +1,150 @@
+
+#include "cuda_modules/cuda_kernels/cuda_kernels.h"
+
+namespace cuvslam::map {
+
+namespace {
+
+// Eviction predicate: the point is "close" to camera `cam` iff its camera-frame
+// z is in (0, max_depth]. No frustum check — a point close but momentarily off
+// the image rect is still worth keeping because small yaw / vibration will
+// routinely bring it back into view.
+inline bool close_to_cam(const Vector3T& p_world, const DepthCameraInfo& cam, float max_depth) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)